pytorch_lightning/metrics/converters.py

```diff
@@ -301,7 +301,7 @@ def gather_all_tensors_if_available(result: Union[torch.Tensor],
 
         world_size = torch.distributed.get_world_size(group)
 
-        gathered_result = world_size * [torch.zeros_like(result)]
+        gathered_result = [torch.zeros_like(result) for _ in range(world_size)]
```

Comment:
Could you add a test to `tests/metrics/test_converters.py` that actually test that this function does what it is expected to do?

Reply:
Yeah, thanks for the advice, I will add a test for this RP.